src/osvr/Util/PlatformConfig.h.in

```diff
@@ -55,33 +55,143 @@
 #endif
 
 /**
- * Platform-specific variables.
+ * \name Operating system tests.
  *
  * Prefer testing for specific compiler or platform features instead of relying
  * on these variables.
  *
+ * Multiple OS defines may be enabled at the same time.
  */
 //@{
-#cmakedefine OSVR_AIX
-#cmakedefine OSVR_ANDROID
-#cmakedefine OSVR_BSDOS
-#cmakedefine OSVR_FREEBSD
-#cmakedefine OSVR_HPUX
-#cmakedefine OSVR_IRIX
-#cmakedefine OSVR_LINUX
-#cmakedefine OSVR_KFREEBSD
-#cmakedefine OSVR_NETBSD
-#cmakedefine OSVR_OPENBSD
-#cmakedefine OSVR_OFS1
-#cmakedefine OSVR_SCO_SV
-#cmakedefine OSVR_UNIXWARE
-#cmakedefine OSVR_XENIX
-#cmakedefine OSVR_SUNOS
-#cmakedefine OSVR_TRU64
-#cmakedefine OSVR_ULTRIX
-#cmakedefine OSVR_CYGWIN
-#cmakedefine OSVR_MACOSX
-#cmakedefine OSVR_WINDOWS
+#if defined(_AIX)
+#define OSVR_AIX 1
+#endif
+
+#if defined(__ANDROID__)
+#define OSVR_ANDROID 1
+#endif
+
+#if defined(__bsdi__)
+#define OSVR_BSDOS 1
+#endif
+
+#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__) || defined(BSD)
+#define OSVR_FREEBSD 1
+#endif
+
+#if defined(hpux) || defined(_hpux) || defined(__hpux)
+#define OSVR_HPUX 1
+#endif
+
+#if defined(sgi) || defined(__sgi)
+#define OSVR_IRIX 1
+#endif
+
+#if defined(linux) || defined(__linux__)
+#define OSVR_LINUX 1
+#endif
+
+#if defined(__FreeBSD_kernel__) && defined(__GLIBC__)
+#define OSVR_KFREEBSD 1
+#endif
+
+#if defined(__NetBSD__)
+#define OSVR_NETBSD 1
+#endif
+
+#if defined(__OpenBSD__)
+#define OSVR_OPENBSD 1
+#endif
+
+#if defined(sco) && defined(_UNIXWARE7)
+#define OSVR_UNIXWARE 1
+#endif
+
+#if defined(sun) || defined(__sun)
+#if defined(__SVR4) || defined(__svr4__)
+#define OSVR_SOLARIS 1
+# else
+#define OSVR_SUNOS 1
+#endif
+#endif
+
+#if defined(__osf) || defined(__osf__)
+#define OSVR_TRU64 1
+#endif
+
+#if defined(ultrix) || defined(__ultrix) || defined(__ultrix__) || (defined(unix) && defined(vax))
+#define OSVR_ULTRIX 1
+#endif
+
+#if defined(__CYGWIN__)
+#define OSVR_CYGWIN 1
+#endif
+
+#if (defined(__APPLE__) && defined(__MACH__))
+#define OSVR_MACOSX 1
+#endif
+
+#if defined(_WIN32) || defined(WIN32) || defined(__NT__) || defined(__WIN32__)
```

Comment:
@JeroMiya `_WIN32` is defined as `1` when the compilation target is 32-bit ARM, 64-bit ARM, x86, or x64. Otherwise, undefined.  `_WIN64` is defined as `1` when the compilation target is 64-bit ARM or x64. Otherwise, undefined. See the [Predefined Macros](https://msdn.microsoft.com/en-us/library/b0084kay.aspx) documentation on MSDN.